client.go

```diff
@@ -422,11 +433,21 @@ func (client *Client) cachedMetadata(topic string, partitionID int32) *Partition
 	return nil
 }
 
-func (client *Client) cachedPartitions(topic string, partitionSet int) []int32 {
+func (client *Client) cachedPartitions(topic string, partitionSet partitionType) []int32 {
 	client.lock.RLock()
 	defer client.lock.RUnlock()
 
 	partitions := client.metadata[topic]
+
+	if partitions == nil {
```

Comment:
It's a bit weird to check `metadata[topic]` for nil, and then unconditionally access `cachedPartitionsResults` (I realize it works because they are always set together, but it reads a little odd)